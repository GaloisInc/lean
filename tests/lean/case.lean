example (xs : list ℕ) : ℕ :=
begin
  induction xs,
  case list.cons {}
end

example (xs : list ℕ) : ℕ :=
begin
  cases xs,
  case list.cons {}
end

example (xs : list ℕ) : ℕ :=
begin
  induction xs,
  case list.cons : x xs {
    cases xs,
    case list.cons : x xs {}
  }
end

open list
example (xs : list ℕ) : ℕ :=
begin
  induction xs,
  case cons {}
end

example (xs : list ℕ) : ℕ :=
begin
  induction xs,
  case list.cons : x xs ih { apply ih },
  case list.nil { apply 0 }
end
