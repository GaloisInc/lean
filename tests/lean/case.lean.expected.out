case.lean:4:2: error: solve1 tactic failed, focused goal has not been solved
state:
case list.cons
xs_hd : ℕ,
xs_tl : list ℕ,
xs_ih : ℕ
⊢ ℕ
case.lean:10:2: error: solve1 tactic failed, focused goal has not been solved
state:
case list.cons
xs_hd : ℕ,
xs_tl : list ℕ
⊢ ℕ
case.lean:18:4: error: solve1 tactic failed, focused goal has not been solved
state:
case list.cons
xs_ih x x : ℕ,
xs : list ℕ
⊢ ℕ
case.lean:26:2: error: solve1 tactic failed, focused goal has not been solved
state:
case list.cons
xs_hd : ℕ,
xs_tl : list ℕ,
xs_ih : ℕ
⊢ ℕ
